### PR TITLE
[@types/lambda-tester] Apply typing based on `aws-lambda` handler type

### PR DESCRIPTION
**Pull request by msuntharesan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/lambda-tester/index.d.ts [hunk 1]
```diff
@@ -1,34 +1,45 @@
...
+        identity(cognitoIdentityId: string, cognitoIdentityPoolId: string): this;
         timeout(seconds: number): this;
         xray(): this;
-        expectSucceed(verifier: Verifier): Promise<any>;
-        expectFail(verifier: Verifier): Promise<any>;
-        expectError(verifier: Verifier): Promise<any>;
-        expectResult(verifier: Verifier): Promise<any>;
-        expectReject(verifier: Verifier): Promise<any>;
-        expectResolve(verifier: Verifier): Promise<any>;
+        expectSucceed<S extends HandlerResult<T> = HandlerResult<T>>(verifier: Verifier<S>): Promise<any>;
```
Comment (RyanCavanaugh) [inline types/lambda-tester/index.d.ts]: These should all just be e.g.
```ts
expectSucceed(verifier: Verifier<HandlerResult<T>>): Promise<any>;`
```

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/lambda-tester/index.d.ts (hunk 1 above)
Comment (msuntharesan) [reply types/lambda-tester/index.d.ts]: This has been resolved in last commit

**Comment by msuntharesan** [comment]
Any update on this PR?

**Comment by msuntharesan** [comment]
Been open for too long. closing this and opening a new one.

**Pull request closed without merging**